### PR TITLE
Create test for useWallet

### LINKED ISSUE #176: Add tests for `useWallet`

### PR DESCRIPTION
**Pull request by gjsyme**
Closes #176 

## Description

Add @testing-library/react-hooks, make use of Provider for context and test that useWallet can instantiate and trigger the request to connect to the correct network.

## 📝 Additional Information

> Anything else you'd want reviewers to know.

Pretty basic tests, but extensible if we want to either deeply mock window.ethereum, make use of another external library (though none jump out as best-choice), or solve this in another way.

### CODE REVIEW

**Comment by etr2460** [comment]
The chromatic run failed because this PR is made from a fork. If you want to, then you could re open the PR from a branch in the repo. that said, since it's only a test, i feel like it would be safe without chromatic running

File: packages/hooks/src/hooks/useWallet.test.tsx [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+let windowSpy;
+
+beforeEach(() => {
+  windowSpy = jest.spyOn(window, 'window', 'get');
+});
+
+afterEach(() => {
+  windowSpy.mockRestore();
+});
```
Comment (etr2460) [inline packages/hooks/src/hooks/useWallet.test.tsx:14]: i'm pretty sure these should go inside the describe block

File: packages/hooks/src/hooks/useWallet.test.tsx [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+describe('useWallet tests', () => {
+  test('should use the wallet', () => {
+    const wrapper = ({ children }) => (
+      <Provider network={1} infuraId={'abc123'}>
```
Comment (etr2460) [inline packages/hooks/src/hooks/useWallet.test.tsx:19]: this is typically how react passes strings as props

```suggestion
      <Provider network={1} infuraId="abc123">
```

File: packages/hooks/src/hooks/useWallet.test.tsx [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+    expect(result.current.connected).toBe(false);
+    expect(typeof result.current.connectWallet).toBe('function');
+  });
+
+  // initially is chainId undefined / requires network per the argument
+  test('should be able to switch networks', () => {
+    const testNetwork = 2;
+    const wrapper = ({ children }) => (
+      <Provider network={testNetwork} infuraId={'abc123'}>
```
Comment (etr2460) [inline packages/hooks/src/hooks/useWallet.test.tsx:33]:
```suggestion
      <Provider network={testNetwork} infuraId="abc123">
```

**Review by etr2460 — COMMENTED** [review]
a few comments but mostly good. thanks for adding tests!

File: packages/hooks/src/hooks/useWallet.test.tsx (hunk 2 above)
Comment (gjsyme) [reply packages/hooks/src/hooks/useWallet.test.tsx:19]: agreed. been using something that has gotten me in the habit of objectifying my strings and i shouldn't.

**Review by etr2460 — APPROVED** [review]
let's do it, thanks for the tests!

**Comment by etr2460** [comment]
@allcontributors add @gjsyme for code

**Pull request merged**